### ISSUE TITLE
Support validate=strict for all component configs

### ISSUE THREAD

**Issue by jpbetz** [kubernetes/kubernetes#127940]
The Kubernetes API supports [strict validation](https://kubernetes.io/blog/2023/04/24/openapi-v3-field-validation-ga/#server-side-field-validation). `kubectl ... --validate='strict'` is also available.

~But there is no way to use this feature with component config files.  Should there be?  How would it be enabled?  Could we define an annotation to enable it?~

EDIT: Turns out that many config files have strict validation enabled, but not all.

/sig api-machinery

**Comment by seans3** [comment]
Component: `kubelet`

TL/DR: The component config for `kubelet` (`KubletConfiguration`) already uses strict validation.

There are two ways a configuration file is read, parsed, and validated for `kubelet`:
  1. Using the `--config` to specify the configuration file
  2. Using the `--config-dir` parameter to specify the directory containing possibly multiple configuration files.

In both cases the files are read from the file system and decoded using a codec created with the `EnableStrict` option.

https://github.com/kubernetes/kubernetes/blob/9cf4f912451009296a50f267d47b484f77ce05e4/pkg/kubelet/kubeletconfig/configfiles/configfiles.go#L50-L52

**Comment by seans3** [comment]
Component: kube-scheduler

TL/DR:The component config for the scheduler (`KubeSchedulerConfiguration`) already uses strict validation.

The codec created to decode and validate the scheduler configuration file is already created enabling strict validation.

https://github.com/kubernetes/kubernetes/blob/41440c8117d0982eccd3c12d240841962101f101/cmd/kube-scheduler/app/options/configfile.go#L42-L44

...and the codec is created here

https://github.com/kubernetes/kubernetes/blob/41440c8117d0982eccd3c12d240841962101f101/pkg/scheduler/apis/config/scheme/scheme.go#L31-L32

**Comment by seans3** [comment]
component: `kube-proxy`

TL/DR: The component config for `kube-proxy` (`KubeProxyConfiguration`) already uses strict validation.

The codec created to decode and validate the `kube-proxy` configuration file is already created enabling strict validation.

https://github.com/kubernetes/kubernetes/blob/41440c8117d0982eccd3c12d240841962101f101/pkg/proxy/apis/config/scheme/scheme.go#L30-L32

...and the codec is used for decoding the configuration file here

https://github.com/kubernetes/kubernetes/blob/41440c8117d0982eccd3c12d240841962101f101/cmd/kube-proxy/app/options.go#L455-L458

**Comment by seans3** [comment]
component: `kube-controller-manager`

TL/DR: The `kube-controller-manger` does not appear to use a configuration file.

The component configuration struct for the `kube-controller-manager` is `KubeControllerManagerConfiguration`. This struct appears to be filled in using flags and default values; there is no configuration file. Additionally, there is a codec created in the `kube-controller-manager` config library (without the `EnableStrict` option), but it does not appear to be referenced anywhere.

https://github.com/kubernetes/kubernetes/blob/f37ab27d9aa5262ebc6249a04ea6acef7b63c76f/pkg/controller/apis/config/scheme/scheme.go#L30-L32

Please let me know if I have missed anything here.

**Comment by seans3** [comment]
component: `kube-apiserver`
sub-component: `AuthorizationConfiguration`

TL/DR: The sub-component for authorization of the API Server is already decoded using `EnableStrict` validation.

The config file for `AuthorizationConfiguration` is read and validated here
https://github.com/kubernetes/kubernetes/blob/74cfa2fd04be671b64aee32e63ebb7272f3b3cfa/pkg/kubeapiserver/options/authorization.go#L223-L239

The data is decoded using a codec here
https://github.com/kubernetes/kubernetes/blob/74cfa2fd04be671b64aee32e63ebb7272f3b3cfa/staging/src/k8s.io/apiserver/pkg/apis/apiserver/load/load.go#L73

...and the codec is created with `EnableStrict` here
https://github.com/kubernetes/kubernetes/blob/74cfa2fd04be671b64aee32e63ebb7272f3b3cfa/staging/src/k8s.io/apiserver/pkg/apis/apiserver/load/load.go#L33

**Comment by seans3** [comment]
component: `kube-apiserver`
sub-component: `EncryptionConfiguration`

TL/DR: The sub-component for encryption within the API Server is already decoded using `EnableStrict` validation.

The config file for `EncryptionConfiguration` is read here
https://github.com/kubernetes/kubernetes/blob/74cfa2fd04be671b64aee32e63ebb7272f3b3cfa/staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go#L211-L216

...and the loaded configuration data is decoded here
https://github.com/kubernetes/kubernetes/blob/74cfa2fd04be671b64aee32e63ebb7272f3b3cfa/staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go#L505-L512

...and the codec used to decode this data is created with `EnableStrict` here
https://github.com/kubernetes/kubernetes/blob/74cfa2fd04be671b64aee32e63ebb7272f3b3cfa/staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go#L134

**Comment by seans3** [comment]
component: `kube-apiserver`
sub-component: `EgressSelectorConfiguration`

TL/DR: The sub-component for egress selector configuration within the API Server is **NOT** decoded using EnableStrict validation. A PR to fix this is required.

The `EgressSelectorConfiguration` is decoded using a simple `yaml.Unmarshal`; it does not do strict validation.
https://github.com/kubernetes/kubernetes/blob/582dcc2acab9417cae3507c0de384c398027e716/staging/src/k8s.io/apiserver/pkg/server/egressselector/config.go#L54-L59

**Comment by seans3** [comment]
component: `kube-apiserver`
sub-component: `AdmissionConfiguration`

TL/DR: The sub-component for admission (including admission plugins) configuration within the API Server is NOT decoded using `EnableStrict` validation. A PR to fix this is required.

**Comment by seans3** [comment]
component: `kube-apiserver`
sub-component: resource quota admission plugin 

TL/DR: The sub-component for the resource quota admission plugin configuration within the API Server is **NOT** decoded using `EnableStrict` validation. A PR to fix this is required.

The configuration for the resource quota admission plugin is read from a file and decoded here
https://github.com/kubernetes/kubernetes/blob/582dcc2acab9417cae3507c0de384c398027e716/staging/src/k8s.io/apiserver/pkg/admission/plugin/resourcequota/config.go#L51-L57

...and the codec decoding the configuration data is created **without** the `EnableStrict` option here
https://github.com/kubernetes/kubernetes/blob/582dcc2acab9417cae3507c0de384c398027e716/staging/src/k8s.io/apiserver/pkg/admission/plugin/resourcequota/config.go#L32

**Comment by seans3** [comment]
component: kube-apiserver
sub-component: event rate limit admission plugin

TL/DR: The sub-component for the event rate limit admission plugin configuration within the API Server is **NOT** decoded using `EnableStrict` validation. A PR to fix this is required.

The configuration for the event rate limit admission plugin is read from a file and decoded here
https://github.com/kubernetes/kubernetes/blob/582dcc2acab9417cae3507c0de384c398027e716/plugin/pkg/admission/eventratelimit/config.go#L52-L57

...and the codec decoding the configuration data is created **without** the `EnableStrict` option here
https://github.com/kubernetes/kubernetes/blob/582dcc2acab9417cae3507c0de384c398027e716/plugin/pkg/admission/eventratelimit/config.go#L32

**Comment by seans3** [comment]
component: `kube-apiserver`
sub-component: pod toleration restriction admission plugin

TL/DR: The sub-component for the pod toleration restriction admission plugin configuration within the API Server is **NOT** decoded using `EnableStrict` validation. A PR to fix this is required.

The configuration for the pod toleration restriction admission plugin is read from a file and decoded here
https://github.com/kubernetes/kubernetes/blob/582dcc2acab9417cae3507c0de384c398027e716/plugin/pkg/admission/podtolerationrestriction/config.go#L53-L58

...and the codec decoding the configuration data is created **without** the `EnableStrict` option here
https://github.com/kubernetes/kubernetes/blob/582dcc2acab9417cae3507c0de384c398027e716/plugin/pkg/admission/podtolerationrestriction/config.go#L33

**Comment by seans3** [comment]
component: `kube-apiserver`
sub-components: admission plugin configurations

OK...it looks like the configuration for just about every admission plugin is **NOT** decoded using a codec with the `EnableStrict` validation option. These admission plugins are within the following directory: 

https://github.com/kubernetes/kubernetes/tree/master/plugin/pkg/admission

There are approximate 23 of these admission plugins in this directory

Example: image policy admission plugin using the `yaml.NewYAMLOrJSONDecoder` instead of a codec for decoding
https://github.com/kubernetes/kubernetes/blob/fe218437e083aa7d59755399d5b08957711a87b1/plugin/pkg/admission/imagepolicy/admission.go#L256-L264

**Comment by seans3** [comment]
component: `kube-apiserver`
sub-component: `TracingConfiguration`
file param: `--tracing-config-file`

TL/DR: The sub-component for the tracing configuration within the API Server is *NOT* decoded using `EnableStrict` validation. A PR to fix this is required.

The configuration for tracing is read from a file and decoded here
https://github.com/kubernetes/kubernetes/blob/023cd33d23067908d08bd21affa43140fb5d7791/staging/src/k8s.io/apiserver/pkg/server/options/tracing.go#L156-L164

...and the codec is created *without* the `EnableStrict` option here
https://github.com/kubernetes/kubernetes/blob/023cd33d23067908d08bd21affa43140fb5d7791/staging/src/k8s.io/apiserver/pkg/server/options/tracing.go#L50
